docs/proposals/20240327-enhancedlivenessprobe.md

```diff
@@ -0,0 +1,189 @@
+
+**Q5: How to protect the application or service from restart encroachment thoroughly?**
++ An algorithm decision unit(workload based), including resilience protection(global switch, blacklist, global limiting rules, etc.), maxUnAvailable protection(PodDisruptionBudget in native or PodUnavailableBudget in kruise), is release in EnhancedLivenessProbeController.
++ A switch will be set in pods, which can close off the detection in emergency.
+
+
+### API Definition
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+  ...
+    // [user defined] using this enhanced livnessProbe, otherwise the native checking
+    apps.kruise.io/using-enhanced-liveness: "true"
+  ...
+  
+  labels:
+  ...
+  ...
+  spec:
+  ...
+    livenessProbe:
+      failureThreshold: 3
+      httpGet:
+        path: /health
+        port: 7002
+        scheme: HTTP
+      initialDelaySeconds: 3000
+      periodSeconds: 100
+      successThreshold: 1
+      timeoutSeconds: 5
+...
+```
+
+
+### Functional Assessment
+| enhanced(this solution)                                                                                                                                                                                  | native                                                                                 | 
+|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------| 
+| :white_check_mark:  specified component detection(nodePodProbe controller in kruise-daemonset)  + global component decision(enhancedLivenessProbeController in kruise-manager) in cluster                | :white_check_mark:  single machine detection + single machine decision(all in kubelet) | 
+| :white_check_mark:  kruiseDaemon/ProbeDetector detection enabled in bypaas,  strong expansibility                                                                                                        | :no_entry: no                                                                          | 
+| :white_check_mark: the global perspective of the application, the maxUnavailable protection policy and the capactiy building of the resilience protection system(Kruise-manager/EnhancedLivenessProbeController) | :no_entry:  no                                                                         | 
+| :white_check_mark: the architecture is easy to expand, without invasive modification of native components, adapt easily to different scaenarios with strong scalability                                  | :no_entry:  strong coupling, poor scalability                                          | 
+
+
+### Other Notes
+● Pod without OwnerReference, no enhancement ability this solution, degradation to community logic, restarting containers when in failure detection.
```

Comment:
enhanced liveness can be working on pod without ownerreference if the flow control logic does not rely on the workload of pod